src/java.base/share/classes/jdk/internal/classfile/impl/AbstractInstruction.java

```diff
@@ -317,7 +317,7 @@ static int size(CodeImpl code, int codeStart, int pos) {
             int pad = ap - (pos + 1);
             int low = code.classReader.readInt(ap + 4);
             int high = code.classReader.readInt(ap + 8);
-            if (high < low || high - low > code.codeLength >> 2) {
+            if (high < low || (long)high - low > code.codeLength >> 2) {
```

Comment:
Maybe `Integer.toUnsignedLong(high - low)` might be clearer?

Reply:
I think it is safer to convert to long beforehand.